src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,63 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (!currentTodo?.userId) {
+      return;
+    }
+
+    getUser(currentTodo.userId)
+      .then(setUser)
+      .finally(() => setIsLoading(false));
+  }, []);
```

Comment:
please try to do it without `isLoading` or fix the current implementation, because now after open modal click - first appears modal, then loader and then modal again, it's definitely not correct